php/libraries/NDB_BVL_Instrument.class.inc

```diff
@@ -1509,11 +1507,15 @@ class NDB_BVL_Instrument extends NDB_Page
 
         $db = \Database::singleton();
 
-        $db->update(
-            $this->table,
-            array('Data_entry_completion_status' => $status),
-            array('CommentID' => $this->getCommentID())
-        );
+        if ($this->jsonData !== true) {
+            // FIXME: Is this even required at all? If it is, we need to
```

Comment:
this is definitely still needed it's the only way an instrument can be set to complete

Reply:
In that case, can we move it to the flag table?

Reply:
This is the only thing blocking the PR at this point